lightning/src/util/persist.rs

```diff
@@ -243,10 +250,886 @@ where
+		let cases = [
+			("deadbeef".to_string(), "cannot parse u64"),
+			("-0000000000000000001".to_string(), "cannot parse u64"),
+		];
+		for (input, expected) in &cases {
+			let got = UpdateName::new(input.to_owned())
+				.expect_err("bad update name accepted")
+				.to_string();
+			assert!(
+				got.contains(*expected),
+				"wrong update name error\nexpected: {}\ngot: {}",
+				expected,
+				got
+			);
+		}
+	}
+
+	#[test]
+	fn test_monitor_from_outpoint_and_as_str_works() {
+		let cases = [
+			(
+				OutPoint {
+					txid: Txid::from_hex(
+						"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef",
+					)
+					.unwrap(),
+					index: 1,
+				},
+				"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef_1",
+			),
+			(
+				OutPoint {
+					txid: Txid::from_hex(
+						"f33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeef",
+					)
+					.unwrap(),
+					index: u16::MAX,
+				},
+				"f33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeef_65535",
+			),
+		];
+		for (input, expected) in &cases {
+			let monitor_name = MonitorName::from(*input);
+			assert_eq!(monitor_name.as_str(), *expected);
+		}
+	}
+
+	#[test]
+	fn test_monitor_from_bad_string_fails() {
+		let cases = [
+			(
+				"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef".to_string(),
+				"No tx index",
+			),
+			(
+				"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef_65536"
+					.to_string(),
+				"Invalid tx index",
+			),
+			(
+				"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef_21".to_string(),
+				"Invalid tx ID",
+			),
+		];
+		for (input, expected) in &cases {
+			let got = MonitorName::new(input.to_owned())
+				.expect_err("bad monitor name accepted")
+				.to_string();
+			assert!(
+				got.contains(*expected),
+				"wrong monitor name error\nexpected: {}\ngot: {}",
+				expected,
+				got
+			);
+		}
+	}
+
+	#[test]
+	fn test_persister_with_real_monitors() {
+		let chanmon_cfgs = create_chanmon_cfgs(4);
+		// Easiest way to get the drop order right is to make the persisters first, then mutate them.
+		let throwaway_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+		let mut persister_0 = default_test_persister(
+			false,
+			throwaway_cfgs[0].keys_manager,
+			throwaway_cfgs[0].keys_manager,
+		);
+		let mut persister_1 = default_test_persister(
+			false,
+			throwaway_cfgs[1].keys_manager,
+			throwaway_cfgs[1].keys_manager,
+		);
+		let mut node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+		persister_0.entropy_source = node_cfgs[0].keys_manager;
+		persister_0.signer_provider = node_cfgs[0].keys_manager;
+		persister_1.entropy_source = node_cfgs[1].keys_manager;
+		persister_1.signer_provider = node_cfgs[1].keys_manager;
+
+		let chain_mon_0 = test_utils::TestChainMonitor::new(
+			Some(&chanmon_cfgs[0].chain_source),
+			&chanmon_cfgs[0].tx_broadcaster,
+			&chanmon_cfgs[0].logger,
+			&chanmon_cfgs[0].fee_estimator,
+			&persister_0,
+			node_cfgs[0].keys_manager,
```

Comment:
Use `&chanmon_cfgs[0].keys_manager`.

Reply:
Sure, but can you tell me why?

Reply:
I guess it isn't strictly necessary, but it is more consistent with the above parameters. A little more intuitive, too, given `chanmon_cfgs` is for initiating the `ChainMonitor`.

Reply:
Got it, okay, thanks.